### PR TITLE
HID-2060: v3 user methods

### PR DESCRIPTION
**Pull request by rupl**
# HID-2060

Big docs PR, plus some other minor fixes that either needed to happen to avoid breaking things, or were small enough that I just didn't feel like waiting:

- HID-2061
- HID-2063

Other tickets filed during the course of this work will be ready for review separately.

I did rearrange the order of some endpoints because I personally felt like it was easier to scan when I read the docs page. All sub-paths of `/api/v3/user/{id}` are grouped together. Because of how `swagger-inline` works it meant I had to move the functions themselves inside `UserController.js` — I looked for a way to apply weights to the definitions but they don't seem to exist. Anyway, it doesn't affect the DX at all IMO, but scanning docs makes a little more sense.

Comparing to https://dev.api-humanitarian-id.ahconu.org/docs/#/ will reveal that many things are missing. To the best of my knowledge, anything omitted in the v3 docs is because it won't be available anyway, so I cut my work by ignoring things we're planning on deprecating.

### CODE REVIEW

File: api/controllers/UserController.js [hunk 1]
```diff
@@ -367,6 +407,75 @@ module.exports = {
...
+   *     in: query
+   *     type: integer
+   *     default: 0
+   *   - name: limit
+   *     description: Maximum number of users to return
+   *     in: query
+   *     type: integer
+   *     default: 50
+   *   - name: fields
+   *     description: Fields to be returned (leave empty to return all)
```
Comment (orakili) [inline api/controllers/UserController.js:432]: Maybe indicate the string format, ex: comma separated list of fields.

**Comment by rupl** [comment]
Regarding TOTP being selectively applied: without any docs or comments, my guess is that it has been applied to "sensitive" fields that directly affect login: generating JWT, password changes, and changing primary email, etc.

Other data (e.g. organization) don't have the potential to disrupt the user's ability to login. I'd say it's fairly normal if we think about GitHub for example: my TOTP or security key is needed at login, and when changing security settings, but changing my profile or other non-essential settings happens without 2FA being required.

**Comment by emmajane** [comment]
90% sure @rupl is correct in terms of field sensitivity. This could be cross-referenced against OICT guidelines to confirm.

File: api/controllers/UserController.js (hunk 1 above)
Comment (rupl) [reply api/controllers/UserController.js:432]: You know, I realized while looking at this that the function shouldn't even exist in v3. We won't be providing the ability to look at another user. Even admins cannot do it anymore.

I'll remove the whole doc block and route.

**Comment by rupl** [comment]
One clue in the codebase is the `security: true` parameter found in logging calls. These are also mandated by OICT and they seem to match up with the same list of operations I mentioned before

**Pull request merged**